lint.go

```diff
@@ -1027,34 +1027,47 @@ func (f *file) lintElses() {
 		if !ok || ifStmt.Else == nil {
 			return true
 		}
-		if ignore[ifStmt] {
-			return true
-		}
-		if elseif, ok := ifStmt.Else.(*ast.IfStmt); ok {
-			ignore[elseif] = true
-			return true
-		}
-		if _, ok := ifStmt.Else.(*ast.BlockStmt); !ok {
-			// only care about elses without conditions
+
+		if _, ok := ignore[ifStmt]; ok {
```

Comment:
It's quite normal to use a map[T]bool containing only true values to represent a set of T, and idiomatic to test for membership of such a set using just m[k].  Please restore the previous check.